### PR TITLE
feat: remove redundant app code-base from 2 test-cases

### LINKED ISSUE #46: testground/app: refactor repetitive steps into reusable components
Introduction 🖖
---
As @Wondertan has correctly mentioned, having redundant code for Celestia App network creation is not a sustainable approach. Thus this issue should address to solving this

What needs to be done 🧐
---
Test-steps like `InitChain`, `CreateKey`, `AddGenAccounts`, `GenTx`, `CollectGenTxs` should be refactored to components in the appkit to make the overall process of bootstrapping an app's network better 
For instance, rn the `RunApp` func is mostly containing the same code base that `InitValidator` has for Application specific instances

#### References 👀 
PR that caused this issue creation: https://github.com/celestiaorg/test-infra/pull/23

### PR DESCRIPTION
**Pull request by Bidon15**
Move the creation of a validator to the common dir
Adjusted the init-validator case to accept the pers-peers param
Move home path to app struct to eliminate passing home var to all methods

This is housekeeping part to start implementing #55 
Closes #46

### CODE REVIEW

File: tests/app-sync/run_validator.go [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+			runenv.RecordFailure(err)
+			return err
+		}
+		go func() {
+			s, err := appkit.GetLatestsBlockSize(net.ParseIP("127.0.0.1"))
+			if err != nil {
+				runenv.RecordMessage("err in last size call, %s", err.Error())
+			}
+
+			runenv.RecordMessage("latest size on iteration %d of the block is - %d", i, s)
```
Comment (vgonkivs) [inline tests/app-sync/run_validator.go:80]: please copy `i` instead of capturing it.

Comment (Bidon15) [reply tests/app-sync/run_validator.go:80]: Fixed. Thanks for the hint 👍

**Review by vgonkivs — APPROVED** [review]

**Review by Wondertan — COMMENTED** [review]
LGTM for this sane refactoring!

File: tests/app-sync/run_validator.go [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+			return err
+		}
+		go func(i int) {
+			s, err := appkit.GetLatestsBlockSize(net.ParseIP("127.0.0.1"))
+			if err != nil {
+				runenv.RecordMessage("err in last size call, %s", err.Error())
+			}
+
+			runenv.RecordMessage("latest size on iteration %d of the block is - %d", i, s)
+		}(i)
```
[the remark this answers is not in the document]
Comment (Bidon15) [reply tests/app-sync/run_validator.go]: After teesting without go func, i see that the issues with 0.35 is not repro here anymore. Removed

**Review by Wondertan — APPROVED** [review]

**Pull request merged**